Add commands for factoid mass-removal

linuxdaemon opened the pull request:
- `forget` now accepts multiple names to delete
- `forgetall` allows a user to clear the entire facts list for a channel

Closes #461

linuxdaemon committed:
Add commands for factoid mass-removal

- `forget` now accepts multiple names to delete
- `forgetall` allows a user to clear the entire facts list for a channel

linuxdaemon committed:
Add tests for clearing factoids

A-UNDERSCORE-D commented on plugins/factoids.py:
```diff
@@ -107,18 +113,42 @@ def remember(text, nick, db, chan, notice, event):
+    :type db: sqlalchemy.orm.Session
+    :type notice: function
+    """
+    for name in text.split():
```
If possible this loop should be further down, as sending a notice per factoid is needlessly spammy

linuxdaemon replied:
I prefer to keep the existing behavior of saying what the value of the fact was before deleting it, which can't be done without a notice per.

A-UNDERSCORE-D replied:
If you're removing multiple factoids, you probably know what is in them. And, you could easily map the factoids and their contents onto one line to be sent to the user.

linuxdaemon replied:
That wouldn't work for longer factoids, which most are.

A-UNDERSCORE-D replied:
It already automatically splits lines that are overly long, you could also send it through a pager easily

linuxdaemon committed:
Add option to raise when web.paste fails

linuxdaemon committed:
Clean up factoid notice spam

Don't send a notice for each factoid removed, do a batch removal and
dump all the old data into a paste

linuxdaemon committed:
Update factoids tests

linuxdaemon committed:
Fix codestyle

A-UNDERSCORE-D commented on cloudbot/util/web.py:
```diff
@@ -59,7 +59,11 @@ def expand(url, service=None):
 
 
-def paste(data, ext='txt', service=DEFAULT_PASTEBIN):
+class NoPasteException(Exception):
```
Why is this in this PR?

linuxdaemon replied:
Because I added it specifically for the use of this plugin (for now)

linuxdaemon committed:
Add tests for error handling in fact removal
